packages/euclid/src/msgs/hook.rs

```diff
@@ -0,0 +1,25 @@
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{to_json_binary, Binary};
+
+use crate::error::ContractError;
+
+#[cw_serde]
+pub struct EuclidReceive {
+    pub data: Binary,
+    // Metadata to be logged into events for some off chain oracle/analytics
+    pub meta: Option<String>,
+}
+
+// This is just a helper to properly serialize the above message
+#[cw_serde]
+pub enum EuclidReceiverMsg {
+    EuclidReceive(EuclidReceive),
+}
+
+impl EuclidReceive {
+    pub fn to_cosmos_msg(&self) -> Result<Binary, ContractError> {
```

Comment:
The function name is a bit misleading since it returns a `Binary` and not a `CosmosMsg`